source/base/mpi.cc

```diff
@@ -448,7 +448,7 @@ namespace Utilities
       MPI_Aint     displacements[] = {0, offsetof(MinMaxAvg, min_index)};
       MPI_Datatype types[]         = {MPI_DOUBLE, MPI_INT};
 
-      ierr = MPI_Type_struct(2, lengths, displacements, types, &type);
+      ierr = MPI_Type_create_struct(2, lengths, displacements, types, &type);
```

Comment:
If the function has been around since 2.0, you'd need to find an MPI installation that predates that version to get into trouble. I think it's probably reasonable that nobody these days has any of these still around. (I don't know when 2.0 was released, but the 2.1 document is dates June 2008 -- more than ten years ago now.)

Reply:
Then, we can say in the documentation that we don't support anything before `MPI-2.0` :smiley: